src/timerfd.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(unused)]
+use std::fs::File;
+use std::mem;
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+use std::ptr;
+use std::time::Duration;
+
+use libc::{self, timerfd_create, timerfd_gettime, timerfd_settime, CLOCK_MONOTONIC, TFD_CLOEXEC};
+
+use crate::errno::{errno_result, Result};
+
+/// A safe wrapper around a Linux timerfd (man 2 timerfd_create).
+pub struct TimerFd(File);
+
+impl TimerFd {
```

Comment:
I haven't check the other docs, but I think it is a good idea to have the documentation split in summary and description.

For new() it would look something like this:
```rust
/// Create a new `TimerFd`.
///
/// The timer is initially disarmed and must be armed by calling `reset()`.
```

Then we could also add links to the structures and methods, but I guess it is optional because it does make the code comments harder to read with the advantage of making the html docs prettier:

```rust
/// Create a new [`TimerFd`](struct.TimerFd.html).
///
/// The timer is initially disarmed and must be armed by calling [`reset`](struct.TimerFd.html#method.reset).
```

P.S. I haven't checked that the documentation links are correct.

Reply:
Let me split the summary and description. The link might be optional and I found not all of the linux man is appended in original codes.

Reply:
Hmm, I don't think I understand, I meant inside links to structures and functions that you are referring to in the comments.
https://rust-lang-nursery.github.io/api-guidelines/documentation.html#prose-contains-hyperlinks-to-relevant-things-c-link

Reply:
@andreeaflorescu Thanks for the guide. I was misunderstood. I'll update a new version today.